### PR TITLE
[ fix #959 ] ES Codegen: Add casts from integer types to double

### LINKED ISSUE #959: Number cast passes type checker but crashes with a type error during JS execution
# Steps to Reproduce
```idris
%foreign "javascript:lambda: (x) => console.log('ok')"
prim__ok : Double -> PrimIO ()

main : HasIO io => io ()
main = do
    primIO $ prim__ok 100
```
Compile to html file and run.

# Expected Behavior
Browser console prints `ok`.

# Observed Behavior
Everything compiles but the browser console prints `ERROR: invalid cast: + Integer + ' -> ' + Double`.

---

# Range

For values of 0 to 99, the program will print `ok` as required.
For values of -0 and less, or 100 and more, the runtime crash described above occurs.

---

Not too sure what to call this issue - I'll rename it when it becomes more clear what the problem is.
This might be related to (or even the same problem as) #802, but the range of values affected seems different.

### PR DESCRIPTION
**Pull request by cypheon**
These casts seemed to be missing in the JS backend. Since the Chez backend does have the casts Int->Double and Integer->Double, I figured it's okay to just add them.
This fixes #959, a test case for the numbers mentioned in the original issue is included in the commit.

### CODE REVIEW

File: src/Compiler/ES/ES.idr [hunk 1]
```diff
@@ -248,6 +248,8 @@ jsOp (Cast CharType IntType) [x] = pure $ toBigInt $ x ++ ".codePointAt(0)"
 jsOp (Cast CharType IntegerType) [x] = pure $ toBigInt $ x ++ ".codePointAt(0)"
 jsOp (Cast DoubleType IntType) [x] = boundedInt 63 $ "BigInt(Math.floor(" ++ x ++ "))"
 jsOp (Cast DoubleType IntegerType) [x] = pure $ "BigInt(Math.floor(" ++ x ++ "))"
+jsOp (Cast IntType DoubleType) [x] = pure $ "parseFloat(" ++ x ++ ")"
+jsOp (Cast IntegerType DoubleType) [x] = pure $ "parseFloat(" ++ x ++ ")"
```
Comment (stepancheg) [inline src/Compiler/ES/ES.idr:252]: Shouldn't this be `Number(x)` instead of print to string and parse?

Comment (cypheon) [reply src/Compiler/ES/ES.idr:252]: The `x` in this function is an _Idris string_ (either the Literal, like "100n", or the name of the JS variable) which represents a _JavaScript BigInt value_. So the actual number is not converted to a string.

Comment (rbarreiro) [reply src/Compiler/ES/ES.idr:252]: But why not use Number instead? It would seam less arcane.

Comment (cypheon) [reply src/Compiler/ES/ES.idr:252]: Yeah that makes sense, I'll update the PR.

Comment (cypheon) [reply src/Compiler/ES/ES.idr:252]: According to MDN, parseFloat _does_ convert its argument to a String first, which is really unneccessary here, so @stepancheg, sorry I misunderstood your initial concern.

**Comment by gallais** [comment]
Thanks!

**Pull request merged**